auth/forward.go

```diff
@@ -0,0 +1,100 @@
+package auth
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/types"
+	"github.com/stretchr/stew/objects"
+)
+
+// Forward the authentication to a external server
+func Forward(forward *types.Forward, w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+
+	client := http.Client{}
+
+	forwardReq, err := http.NewRequest("GET", forward.Address, nil)
+	if err != nil {
+		log.Debugf("Error calling %s. Cause %s", forward.Address, err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if forward.ForwardAllHeaders {
+		forwardReq.Header = r.Header
+	}
+
+	forwardReq.Header.Add("Accept", "application/json")
```

Comment:
I'm not sure that this header still fits as if the auth server rejects the request with a body, that is returned to the client, that could be xml or whatever the auth server deems appropriate.